### ISSUE TITLE
🐛 Bug Report: Template review page should not invent the parameter names

### ISSUE THREAD

**Issue by dmartinol** [backstage/backstage#23794]
### 📜 Description

When you create a software template and provide some input parameters, the Review page shows a summary table of the provided inputs with a left column that is not reflecting the parameter labels displayed in the previous pages.

FYI @masayag @pkliczewski @gciavarrini

### 👍 Expected behavior

The parameter name in the Review page should be the same as the one in the configuration screen.

### 👎 Actual Behavior with Screenshots

E.g., if we define an input parameter as:
```yaml
                argocdNamespace:
                  title: ArgoCD Namespace
                  type: string
```
the configuration screen displays the proper label from the parameter title as `ArgoCD Namespace`, while the Review page, for the same parameter, displays `Argocd Namespace`.
Our guess is that this page is trying to apply a heuristic to compute a human readable label, by adding a space before any uppercase character in the property name `argocdNamespace`, and always capitalize the first letter.
Anyway, whatever the transformation logic adopted here, the question is why this page is not using the original labels instead.

### 👟 Reproduction steps

Create a template having some input parameters and compare the labels in the Review screens against those in the previous pages.

### 📃 Provide the context for the Bug.

We believe that providing consistent information in all the screens should be a must.

### 🖥️ Your Environment

Applies to any environment

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by benjdlambert** [comment]
Ah, yep I see it now.

We use the `StructuredMetadataTable` from `@backstage/core-components` which does some formatting things as it looks like we just take `key/value` from the `formData` rather than the title.

https://github.com/backstage/backstage/blob/ce8a6bd6f8ff83ec70e7831ca55798b178050c53/packages/core-components/src/components/StructuredMetadataTable/StructuredMetadataTable.tsx#L176-L182

I think what we want to do is used the `definitionInSchema` to pull the `title` part from the original JSON Schema to use as the `key` instead.

https://github.com/backstage/backstage/blob/d6406e5614d31759826a52239b8da8254f377665/plugins/scaffolder-react/src/next/components/ReviewState/ReviewState.tsx#L41-L44

I think this should be `definitionInSchema?.title ?? key`

https://github.com/backstage/backstage/blob/d6406e5614d31759826a52239b8da8254f377665/plugins/scaffolder-react/src/next/components/ReviewState/ReviewState.tsx#L73

**Comment by benjdlambert** [comment]
@dmartinol do you want to whip up a PR with the suggested change?

**Comment by pkliczewski** [comment]
@batzionb please take a look

**Comment by benjdlambert** [comment]
@dmartinol @batzionb are there any updates here? :pray:

**Comment by benjdlambert** [comment]
Closing as fixed in #26196

**Comment by batzionb** [comment]
@benjdlambert 
Thanks for the fix!
